apps/application-system/form/src/components/Authenticator.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { FC, useEffect, useState } from 'react'
+import useAuth from '../hooks/useAuth'
+import AuthenticationLoading from './AuthenticationLoading/AuthenticationLoading'
+
+export const Authenticator: FC = ({ children }) => {
+  const { userInfo, userInfoState, signInUser } = useAuth()
+  const [pendingAuthentication, setPending] = useState(false)
+
+  useEffect(() => {
+    if (
+      !pendingAuthentication &&
+      userInfo === undefined &&
+      userInfoState === 'passive'
+    ) {
+      setPending(true)
+      signInUser()
+        .then(() => {
+          setPending(false)
+        })
+        .catch(() => {
+          setPending(false)
+        })
+    }
+  }, [pendingAuthentication, userInfo, userInfoState, signInUser])
+
+  return <>{userInfo ? children : <AuthenticationLoading />}</>
```

Comment:
For future PR: Would be nice if we had an error screen for the catch, otherwise they would just think it's loading and stuck.